10_datastructures_lab/01.py

```diff
@@ -1,10 +1,15 @@
-"""
-Write a program that takes two strings
-from the user and checks if they represent
-a valid user name.
-Valid users and passwords:
-    apple => red
-    lettuce => green
-    lemon => yellow
-    orange => orange
-"""
+""" Return IPs from hosts file per input machine name(s) """
+import sys
+from collections import defaultdict
+# Read input args
+host_file_path, pc_names = sys.argv[1], sorted(sys.argv[2:])
+# Read hosts into a dictionary
+dict_hosts = defaultdict(str)

```

Comment:
Nice!!


Regards,

Nir Naor
Video Analytics R&D Team Leader
Video & Situation Intelligence Solutions
Verint Systems Inc.
Phone:  +972-9-9622512
Cell:      +972-52-3334029
Email:   Nir.Naor@verint.com<mailto:Nir.Naor@verint.com>
Web:     www.verint.com<http://www.verint.com/>

From: ynonp [mailto:notifications@github.com]
Sent: Wednesday, July 13, 2016 4:49 PM
To: ynonp/python-examples-verint-2016-07
Cc: Naor, Nir; Author
Subject: Re: [ynonp/python-examples-verint-2016-07] ToCode Solution: תרגול מבני נתונים (#108)


In 10_datastructures_lab/01.py<https://github.com/ynonp/python-examples-verint-2016-07/pull/108#discussion_r70627873>:

> -Write a program that takes two strings

> -from the user and checks if they represent

> -a valid user name.

> -Valid users and passwords:

> -    apple => red

> -    lettuce => green

> -    lemon => yellow

> -    orange => orange

> -"""

> +""" Return IPs from hosts file per input machine name(s) """

> +import sys

> +from collections import defaultdict

> +# Read input args

> +host_file_path, pc_names = sys.argv[1], sorted(sys.argv[2:])

> +# Read hosts into a dictionary

> +dict_hosts = defaultdict(str)

Cool! And the nice thing with defaultdict is you can tell it to return any default value (not just the empty string) using lambda notation. With this one:

dict_hosts = defaultdict(lambda: "No IP")

You can simplify the print loop:

for pc_name in pc_names: print "%-*s: %s" % (max_length, pc_name, dict_hosts[pc_name])

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/ynonp/python-examples-verint-2016-07/pull/108/files/bb139a3decc13dea933fffa3965bd1485f056b28#r70627873>, or mute the thread<https://github.com/notifications/unsubscribe/ATWluvOgYEG9yklHUNSN6ZSYOwAkXFkQks5qVOzngaJpZM4JK1k0>.

-- 
This electronic message may contain proprietary and confidential 
information of Verint Systems Inc., its affiliates and/or subsidiaries.
The information is intended to be for the use of the individual(s) or
entity(ies) named above. If you are not the intended recipient (or 
authorized to receive this e-mail for the intended recipient), you may not 
use, copy, disclose or distribute to anyone this message or any information 
contained in this message. If you have received this electronic message in 
error, please notify us by replying to this e-mail.